libsolidity/analysis/ConstantEvaluator.h

```diff
@@ -38,8 +38,9 @@ class TypeChecker;
 class ConstantEvaluator: private ASTConstVisitor
 {
 public:
-	ConstantEvaluator(Expression const& _expr, ErrorReporter& _errorReporter):
-		m_errorReporter(_errorReporter)
+	ConstantEvaluator(Expression const& _expr, ErrorReporter& _errorReporter, size_t _newDepth):
```

Comment:
Could have default value for `_newDepth = 0` so that `ReferencesResolver` doesn't need to be updated.